shell/platform/embedder/embedder.cc

```diff
@@ -543,28 +543,44 @@ void PopulateSnapshotMappingCallbacks(const FlutterProjectArgs* args,
   };
 
   if (flutter::DartVM::IsRunningPrecompiledCode()) {
+    bool provided_data_buffer = false;
     if (SAFE_ACCESS(args, vm_snapshot_data, nullptr) != nullptr) {
+      provided_data_buffer = true;
       settings.vm_snapshot_data = make_mapping_callback(
           args->vm_snapshot_data, SAFE_ACCESS(args, vm_snapshot_data_size, 0));
     }
 
     if (SAFE_ACCESS(args, vm_snapshot_instructions, nullptr) != nullptr) {
+      provided_data_buffer = true;
       settings.vm_snapshot_instr = make_mapping_callback(
           args->vm_snapshot_instructions,
           SAFE_ACCESS(args, vm_snapshot_instructions_size, 0));
     }
 
     if (SAFE_ACCESS(args, isolate_snapshot_data, nullptr) != nullptr) {
+      provided_data_buffer = true;
       settings.isolate_snapshot_data = make_mapping_callback(
           args->isolate_snapshot_data,
           SAFE_ACCESS(args, isolate_snapshot_data_size, 0));
     }
 
     if (SAFE_ACCESS(args, isolate_snapshot_instructions, nullptr) != nullptr) {
+      provided_data_buffer = true;
       settings.isolate_snapshot_instr = make_mapping_callback(
           args->isolate_snapshot_instructions,
           SAFE_ACCESS(args, isolate_snapshot_instructions_size, 0));
     }
+
+    if (SAFE_ACCESS(args, application_library_path, nullptr) != nullptr) {
+      if (provided_data_buffer) {
+        LOG_EMBEDDER_ERROR(
```

Comment:
Missing `return` in case of this error.

Reply:
You can add a test for this as well (as discussed in `embedder_unittests.cc`).

Reply:
If I wanted to actually launch the embedder with an ELF binary, where would I need to configure this?